docs/api_template.rst

```diff
@@ -0,0 +1,60 @@
+namespace::function
+###################
+
+Defined in header: :code:`Namespace_function.hpp`
+
+.. code:: c++
+
+  template <class execution_space, class input_type, class output_type>
+  void function(const execution_space& space, const input_type& in, const output_type& out);
+
+  template <class input_type, class output_type>
+  void function(const input_type& in, const output_type& out);
+
+Explain what the function does and what each overall specificities are.
+
+Parameters
+==========
+
+:space: execution space instance
+
+:in: input
+
+:out: output
+
+Type Requirements
+-----------------
+
+- `execution_space` must be a Kokkos `execution space <https://kokkos.org/kokkos-core-wiki/API/core/execution_spaces.html>`_
+
+- `input_type` must be a Kokkos `View <https://kokkos.org/kokkos-core-wiki/API/core/view/view.html>`_
+
+  - ``Kokkos::SpaceAccessibility<execution_space, typename input_type::memory_space>::accessible``
+  - ...
+
+- `output_type` must be a Kokkos `View <https://kokkos.org/kokkos-core-wiki/API/core/view/view.html>`_ that satisfies
+
+  - ``Kokkos::SpaceAccessibility<execution_space, typename output_type::memory_space>::accessible``
+  - ...
+
+Example
+=======
+
+.. code:: cppkokkos
+
+  #include "Kokkos_Core.hpp"
+  #include "Namespace_function.hpp"
```

Comment:
```suggestion
  #include <Kokkos_Core.hpp>
  #include <namespace_function.hpp>
```

Reply:
I disagree since despite all our serious we are not a standard library: https://stackoverflow.com/questions/21593/what-is-the-difference-between-include-filename-and-include-filename

Reply:
Hmm, that is discussing. 
Many Kokkos and Kokkos-kernels examples are written in my way.
Can we get an opinion from someone else?